app/models/piece.rb

```diff
@@ -42,6 +55,12 @@ def obstructed?(destination_x, destination_y)
 
   private
 
+  def remove_piece(destination_x, destination_y)
+    destination_piece = Piece.find_by column_coordinate: destination_x, row_coordinate: destination_y
+    destination_piece[:is_on_board?] = false
+    destination_piece
```

Comment:
shouldn't this be:
destination_piece.update(is_on_board?: false) or
destination_piece.update_attributes(is_on_board?: false)

I guess this is worth a try